src/praisonai/praisonai/persistence/conversation/_sql_base.py

```
@@ -0,0 +1,459 @@
+                params.append(user_id)
+            if agent_id:
+                conditions.append(f"agent_id = {self._param}")
+                params.append(agent_id)
+            
+            where_clause = " WHERE " + " AND ".join(conditions) if conditions else ""
+            sql = f"""
+                SELECT * FROM {self.sessions_table} 
+                {where_clause}
+                ORDER BY updated_at DESC
+                LIMIT {limit} OFFSET {offset}
+            """
```

Ruff (0.15.15)

[error] Possible SQL injection vector through string-based query construction

(S608)